src/ott/problems/linear/potentials.py

```diff
@@ -258,3 +264,125 @@ def epsilon(self) -> float:
 
   def tree_flatten(self) -> Tuple[Sequence[Any], Dict[str, Any]]:
     return [self._f, self._g, self._prob, self._f_xx, self._g_yy], {}
+
+
+def plot_ot_map(
+    learned_potentials: DualPotentials,
+    source: jnp.ndarray,
+    target: jnp.ndarray,
+    inverse: bool = False,
+    ax: Optional[matplotlib.axes.Axes] = None,
+    **kwargs: Any,
+) -> Tuple[matplotlib.figure.Figure, matplotlib.axes.Axes]:
+  """Plot data and learned optimal transport map.
+
+  Args:
+    learned_potentials: the potentials object for the map
+    source: samples from the source measure
+    target: samples from the target measure
+    inverse: if the inverse map from the potentials
+      should be used
+    ax: axis to add the plot to
+
+  Returns: matplotlib figure and axis with the plots
+  """
+
+  def draw_arrows(a: np.ndarray, b: np.ndarray) -> None:
+    ax.arrow(
+        a[0], a[1], b[0] - a[0], b[1] - a[1], color=[0.5, 0.5, 1], alpha=0.3
+    )
+
+  grad_state_s = learned_potentials.transport(source, forward=not inverse)
+
+  if ax is None:
+    fig = plt.figure(facecolor="white")
+    ax = fig.add_subplot(111)
+  else:
+    fig = ax.get_figure()
+
+  if not inverse:
+    label_src, label_tgt = 'source', 'target'
+    label_transport = r"$\nabla f(source)$"
+  else:
+    label_src, label_tgt = 'target', 'source'
+    label_transport = r"$\nabla g(source)$"
+
+  ax.scatter(
+      source[:, 0],
+      source[:, 1],
+      color="#1A254B",
+      alpha=0.5,
+      label=label_src,
+      **kwargs,
+  )
+  ax.scatter(
+      target[:, 0],
+      target[:, 1],
+      color="#A7BED3",
+      alpha=0.5,
+      label=label_tgt,
+      **kwargs,
+  )
+  ax.scatter(
+      grad_state_s[:, 0],
+      grad_state_s[:, 1],
+      color="#F2545B",
+      alpha=0.5,
+      label=label_transport,
+      **kwargs,
+  )
+
+  fig.legend(ncol=3, loc="upper center")
```

Comment:
Re your comment, I think I'd prefer the legend inside with defaults (no `ncol`, etc.). If you want, you can add `legend_kwargs: Mapping[str, Any]` for more control.

Reply:
I don't like how it moves around the placement of the legend depending on the contents, are you okay with it?

![image](https://user-images.githubusercontent.com/707462/217888336-ba8d3838-8a64-4ce3-83bf-90ad83fc8e9f.png)

Reply:
I don't mind this, but would re-consider exposing the legend kwargs.

Reply:
Hmm, are you ok with `**kwargs` of `plot_ot_map` being the kwargs for scatter and having `legend_kwargs` separate? We could also make them both dicts as `legend_kwargs` and `scatter_kwargs`.

Reply:
Both sound good, will leave it up to you.